corehq/apps/accounting/models.py

```diff
@@ -2004,6 +2004,8 @@ class InvoiceBase(models.Model):
     is_hidden_to_ops = models.BooleanField(default=False)
     last_modified = models.DateTimeField(auto_now=True)
 
+    note = models.TextField(null=True)
```

Comment:
It would be great to remove `null=True` and instead have `default=''`. Nulls in database columns make comparisons more tricky and in general are a common source of bugs.

The migration may need to use custom `ALTER TABLE ... ADD COLUMN ... NOT NULL DEFAULT ''` SQL since I think Django does not do database-level column defaults (it implements them in Python code), and you can't add a `NOT NULL` column without a default value to an existing table if it has any rows.

Reply:
Thank you Daniel. Fixed here: ea7394da3ccfbbd476b80caf9ff218f2c39a244e
I used `makemigrations` command to generate the migration code, and then validated by `sqlmigrate` command:
```
./manage.py sqlmigrate accounting 0086        
corehq.util.log.HqAdminEmailHandler logger is being changed to logging.StreamHandler
BEGIN;
--
-- Add field note to customerinvoice
--
ALTER TABLE "accounting_customerinvoice" ADD COLUMN "note" text DEFAULT '' NOT NULL;
ALTER TABLE "accounting_customerinvoice" ALTER COLUMN "note" DROP DEFAULT;
--
-- Add field note to invoice
--
ALTER TABLE "accounting_invoice" ADD COLUMN "note" text DEFAULT '' NOT NULL;
ALTER TABLE "accounting_invoice" ALTER COLUMN "note" DROP DEFAULT;
--
-- Add field note to wireinvoice
--
ALTER TABLE "accounting_wireinvoice" ADD COLUMN "note" text DEFAULT '' NOT NULL;
ALTER TABLE "accounting_wireinvoice" ALTER COLUMN "note" DROP DEFAULT;
COMMIT;
```
This output looks good to me. So Django does do database-level column defaults now? Seems like [they add this new feature 4 months ago](https://code.djangoproject.com/ticket/470#no1)

Reply:
Oh, excellent! I think the feature you linked is slightly different, and [will be released as part of Django 5.0](https://github.com/django/django/commit/7414704e88d73dafbcfbb85f9bc54cb6111439d3#diff-6b551f383fcc71f808a17d213b0f925ce9655ce58f18cb7ca3a6052eb3b72b8b). However, it does look like the Django 3.2 knows how to use a default when adding a `NOT NULL` constraint to a column, which is all we need here.

Reply:
Thank you for linking me to the `releases/5.0.txt`! I was also wondering when will this be released and don't know where should I look at.